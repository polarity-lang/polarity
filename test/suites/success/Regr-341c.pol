data Bool { T, F }

data Eq(x y: Bool) {
    Refl(x: Bool): Eq(x, x)
}

codata Fun {
    .ap(x: Bool): Bool
}

def Bool.neg: Bool {
    T => F,
    F => T,
}

-- | Annotations can occur in neutral expressions
#[transparent]
let foo: Fun {
    comatch {
        .ap(x) =>(x: Bool).neg
    }
}

let bar: Eq(foo.ap(T), F) {
    Refl(F)
}
