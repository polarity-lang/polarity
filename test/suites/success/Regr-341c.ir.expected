def .neg {
    T => F,
    F => T
}

let foo { comatch { .ap(x) => x.neg } }

let bar { Refl(F) }
